### PR TITLE
Nette 3.0 version

### PR DESCRIPTION
**Pull request by haltuf**
Hello, sending PR to make Kdyby/Translation compatible with Nette v3.0

This PR is conditional of approving kdyby/monolog PR in here: https://github.com/Kdyby/Monolog/pull/35

To make all tests work, I made temporary change (in https://github.com/Kdyby/Translation/commit/d092f0eb1b53ab5dd81e7f22873b8228fed1b941) to composer.json to make the package rely on the branch consisting of the content of this PR. This should be reverted/edited, once kdyby/monolog has a version compatible with Nette 3.0

The code should pass all the tests (at least it does on my travis, coveralls etc.), which unfortunatelly does not guarantee, that it will work, as the code coverage is quite low. Please help me with testing before approving this PR.

Anyone wishing to contribute to this PR, contact me and I can give you access.

### CODE REVIEW

File: src/Caching/PhpFileStorage.php [hunk 1]
```diff
@@ -23,13 +23,20 @@ class PhpFileStorage extends \Nette\Caching\Storages\FileStorage implements \Net
 	 */
 	public $hint;
 
+	/**
+	 * Additional cache structure
+	 */
+	private const
+		FILE = 'file',
+		HANDLE = 'handle';
```
Comment (enumag) [inline src/Caching/PhpFileStorage.php]: use separate const keywords

Comment (haltuf) [reply src/Caching/PhpFileStorage.php]: done, with several other fixes.

**Comment by haltuf** [comment]
Temporary dependency on my branch of kdyby/monolog removed, replaced with kdyby/monolog:^2.0, that was just released.

**Comment by martinvenus** [comment]
Kdyby/translation is the only dependency preventing our projects to be upgraded to Nette 3. We are testing our project with haltuf/Translation and it seems ok. Can I help somehow to merge this PR?

**Comment by haltuf** [comment]
This PR is imho ready to be merged, I also didn't find any problem in real life nette 3.0 use case. The only possible problem is that I'm on out of office now, and won't be able to prepare hotfix (if needed) in the upcoming 2 weeks.

**Pull request merged**